<launch>
    <!--1、启动乌龟GUI与键盘控制节点-->
    <node pkg = "turtlesim" type = "turtlesim_node" name = "turtle1" output = "screen" />
    <node pkg = "turtlesim" type = "turtle_teleop_key" name = "key" output = "screen" />
    <!--2、生成新的乌龟节点-->
    <node pkg = "step1_new_turtle" type = "new_turtle" name = "turtle2" output = "screen" />

    <!--3、需要启动两只小乌龟相对于世界的坐标关系的发布-->
    <!--
                基本实现思路：
                            1、节点只编写一个
                            2、该节点需要启动两次
                            3、节点启动时动态传参：第一次turtle1 第二次turtle2
    -->
    <node pkg = "step2_pub_turtlestamp" type = "pub_turtlestamp" name = "pub1" args = "turtle1" output = "screen" />
    <node pkg = "step2_pub_turtlestamp" type = "pub_turtlestamp" name = "pub2" args = "turtle2" output = "screen" />

    <!--4、需要订阅 turtle1与turtle2 相对于世界坐标系的坐标消息，
    并转换成turtle1相对于turtle2的坐标关系，并生成速度信息-->
    <node pkg = "step3_control_turtle2" type = "control_turtle2" name = "control"  output = "screen" />


</launch>